### PR TITLE
codeintel: Add repository summary resolvers

### PR DESCRIPTION
**Pull request by efritz**
This PR adds additional resolvers to help determine a repo state w.r.t. code intelligence. These resolvers add ways to get timestamps of last activity for auto-indexing and upload expiration scans, as well as a summary of upload and index records for each unique index/root pair. This will let us see if a particular portion of a repo is currently or consistently failing.

This PR is currently blocking progress on badge work in #33153.

## Test plan

Additional unit tests in the PR.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/codeintel.graphql [hunk 1]
```diff
@@ -567,6 +572,69 @@ extend type GitBlob {
...
+type LSIFUploadsWithRepositoryNamespace {
+    """
+    The root of all uploads in the associated connection.
+    """
+    root: String!
+
+    """
+    The indexer used to produce the each upload in the associated connection.
+    """
+    indexer: String!
```
Comment (Strum355) [inline cmd/frontend/graphqlbackend/codeintel.graphql:613]: Would using the Indexer type here be useful or nah?

File: cmd/frontend/graphqlbackend/codeintel.graphql [hunk 2]
```diff
@@ -567,6 +572,69 @@ extend type GitBlob {
...
+    """
+    lastUploadRetentionScan: DateTime
+
+    """
+    The last time this repository was considered for auto-indexing job scheduling.
+    """
+    lastIndexScan: DateTime
+}
+
+type LSIFUploadsWithRepositoryNamespace {
```
Comment (Strum355) [inline cmd/frontend/graphqlbackend/codeintel.graphql:604]: type comment :blobpls:

File: cmd/frontend/graphqlbackend/codeintel.graphql (hunk 1 above)
Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.graphql:616]: Updating.

Comment (efritz) [reply cmd/frontend/graphqlbackend/codeintel.graphql]: ...and updated.

**Pull request merged**